### ISSUE TITLE
VNC login password prompt not working on LXD containers

### ISSUE THREAD

**Issue by xorel** [OpenNebula/one#3019]
**Description**
VNC is not working for LXD deployments.

**To Reproduce**
Install OpenNebula 5.8 on Ubuntu 18.04, deploy LXD host. Instantiate Centos7 and click VNC button

**Expected behavior**
Console opens

**Details**
 - Affected Component: Sunstone
 - Hypervisor: LXD
 - Version: 5.8

**Additional context**
When I put back the '/bin/sh' instead of '/bin/login' it starts working (but without login/password).

## Progress Status
- [ ] Branch created 
- [ ] Code committed to development branch
- [ ] Testing - QA
- [ ] Documentation 
- [ ] Release notes - resolved issues, compatibility, known issues
- [ ] Code committed to upstream release/hotfix branches
- [ ] Documentation committed to upstream release/hotfix branches

**Comment by xorel** [comment]
OK, but it does not work for me on Ubuntu either (although the error is different). I get the login prompt but not the password prompt. I use "Ubuntu 18.04 - KVM" from the Market place. 

Do you know what could be the problem?

![ubuntu-vnc](https://user-images.githubusercontent.com/1098246/53748287-59491b00-3ea5-11e9-80be-0612bcf3f561.png)

**Comment by dann1** [comment]
This is odd, vanilla containers aren't working either.

```
root@ubuntu1804-context-lxd-fdaf5-0:~# lxc list
+------------------+---------+------------------------+------+------------+-----------+
|       NAME       |  STATE  |          IPV4          | IPV6 |    TYPE    | SNAPSHOTS |
+------------------+---------+------------------------+------+------------+-----------+
| one-10           | RUNNING |                        |      | PERSISTENT | 0         |
+------------------+---------+------------------------+------+------------+-----------+
| one-9            | RUNNING | 192.168.150.105 (eth5) |      | PERSISTENT | 0         |
|                  |         | 192.168.150.104 (eth4) |      |            |           |
|                  |         | 192.168.150.103 (eth3) |      |            |           |
|                  |         | 192.168.150.102 (eth2) |      |            |           |
|                  |         | 192.168.150.101 (eth1) |      |            |           |
|                  |         | 192.168.150.100 (eth0) |      |            |           |
+------------------+---------+------------------------+------+------------+-----------+
| profound-crappie | RUNNING |                        |      | PERSISTENT | 0         |
+------------------+---------+------------------------+------+------------+-----------+
root@ubuntu1804-context-lxd-fdaf5-0:~# lxc exec profound-crappie login
profound-crappie login: root

Login incorrect
profound-crappie login: 
```

Addin strange note, writing `\` next to root in the user prompt allows the password prompt to come


```
root@ubuntu1804-61495:~# lxc list
+-----------------+---------+------+------+------------+-----------+
|      NAME       |  STATE  | IPV4 | IPV6 |    TYPE    | SNAPSHOTS |
+-----------------+---------+------+------+------------+-----------+
| amazing-cricket | RUNNING |      |      | PERSISTENT | 0         |
+-----------------+---------+------+------+------------+-----------+
root@ubuntu1804-61495:~# lxc exec amazing-cricket login
amazing-cricket login: root

Login incorrect
amazing-cricket login: root\
Password: 

Login incorrect
amazing-cricket login:
```

I just tested this on a vanilla LXD setup in a standard ubuntu-1804 KVM VM

**Comment by xorel** [comment]
Side note: when switching to '/bin/bash' it works for Centos and also Ubuntu containers but only on 18.04 hypervisor, on Ubuntu 18.10 it still doesn't work.

**Comment by dann1** [comment]
The snap patch skipped [the VNC code](https://github.com/dann1/one/blob/b4af541fe954267baf64cbc0bc9dc8d8b978b05d/src/vmm_mad/remotes/lib/lxd/opennebula_vm.rb#L390). The lxc command requires sudo to run on 18.10

**Comment by rsmontero** [comment]
This is going to be addressed by an OpenNebula/addon-context-linux#150 Closing this one.
